source/MaterialXRender/Handlers/ImageHandler.cpp

```diff
@@ -1,38 +1,78 @@
+#include <MaterialXCore/Types.h>
+#include <MaterialXGenShader/Util.h>
 #include <MaterialXRender/Handlers/ImageHandler.h>
 #include <cmath>
 
 namespace MaterialX
 {
+bool ImageHandler::saveImage(const std::string& fileName,
+                            const ImageDesc &imageDesc)
+{
+    for (auto loader : _imageLoaders)
+    {
+        if (loader->supportsExtension(getFileExtension(fileName)))
+        {
+            return loader->saveImage(fileName, imageDesc);
+        }
+    }
+    return false;
+}
 
-bool ImageHandler::createDefaultImage(unsigned int& width,
-                                      unsigned int& height,
-                                      unsigned char** buffer)
+bool ImageHandler::getImage(std::string& fileName,
+                             ImageDesc &imageDesc, 
+                             bool /*generatateMipMaps*/)
 {
-    // Create a ramp texture as the default
-    //
-    const unsigned int imageSize = 256;
-    unsigned int middle = imageSize / 2;
-    *buffer = new unsigned char[imageSize * imageSize * 4];
-    unsigned char* pixel = *buffer;
-    for (unsigned int i = 0; i<imageSize; i++)
+    for (auto loader : _imageLoaders)
     {
-        for (unsigned int j = 0; j<imageSize; j++)
+        if (loader->supportsExtension(getFileExtension(fileName)))
```

Comment:
Extension check allows branching to different image loaders. Jonathan S. would like to try out the stb library for additional codecs beyond exr as this is pretty restrictive.